src/tests/presets/test_eth_account.cairo

```diff
@@ -3,38 +3,42 @@ use openzeppelin::account::interface::ISRC6_ID;
 use openzeppelin::account::utils::secp256k1::{
     DebugSecp256k1Point, Secp256k1PointSerde, Secp256k1PointPartialEq
 };
-use openzeppelin::account::utils::signature::EthSignature;
 use openzeppelin::introspection::interface::ISRC5_ID;
 use openzeppelin::presets::EthAccountUpgradeable;
+use openzeppelin::presets::interfaces::eth_account::{
+    EthAccountUpgradeableABISafeDispatcher, EthAccountUpgradeableABISafeDispatcherTrait
+};
 use openzeppelin::presets::interfaces::{
     EthAccountUpgradeableABIDispatcher, EthAccountUpgradeableABIDispatcherTrait
 };
+use openzeppelin::tests::account::ethereum::common::EthAccountSpyHelpers;
 use openzeppelin::tests::account::ethereum::common::{
-    assert_only_event_owner_added, assert_event_owner_removed
-};
-use openzeppelin::tests::account::ethereum::common::{
-    deploy_erc20, get_points, NEW_ETH_PUBKEY, SIGNED_TX_DATA, SignedTransactionData
+    deploy_erc20, SIGNED_TX_DATA, SignedTransactionData, get_accept_ownership_signature
 };
-use openzeppelin::tests::mocks::eth_account_mocks::SnakeEthAccountMock;
-use openzeppelin::tests::upgrades::common::assert_only_event_upgraded;
+use openzeppelin::tests::upgrades::common::UpgradeableSpyHelpers;
+use openzeppelin::tests::utils::constants::secp256k1::KEY_PAIR;
 use openzeppelin::tests::utils::constants::{
-    CLASS_HASH_ZERO, ETH_PUBKEY, SALT, ZERO, RECIPIENT, QUERY_VERSION, MIN_TRANSACTION_VERSION
+    CLASS_HASH_ZERO, ETH_PUBKEY, NEW_ETH_PUBKEY, SALT, ZERO, RECIPIENT, QUERY_VERSION,
+    MIN_TRANSACTION_VERSION
 };
 use openzeppelin::tests::utils;
 use openzeppelin::token::erc20::interface::IERC20DispatcherTrait;
 use openzeppelin::utils::selectors;
 use openzeppelin::utils::serde::SerializedAppend;
+use snforge_std::{
+    cheat_signature_global, cheat_transaction_version_global, cheat_transaction_hash_global,
+    start_cheat_caller_address
+};
+use snforge_std::{spy_events, test_address};
+use starknet::ClassHash;
+use starknet::SyscallResultTrait;
 use starknet::account::Call;
 use starknet::contract_address_const;
-use starknet::testing;
-use starknet::{ContractAddress, ClassHash};
-
-fn CLASS_HASH() -> felt252 {
-    EthAccountUpgradeable::TEST_CLASS_HASH
-}
+use starknet::secp256_trait::Secp256Trait;
+use starknet::secp256k1::Secp256k1Point;
 
 fn V2_CLASS_HASH() -> ClassHash {
-    SnakeEthAccountMock::TEST_CLASS_HASH.try_into().unwrap()
+    utils::declare_class("SnakeEthAccountMock").class_hash
 }
```

Comment:
This way a function named in a constant-naming way implicitly declares a class. Calling this function seems fine, but would actually lead to a runtime error. I think it would be better to give it a name that explicitly states that a new class is to be declared, `declare_v2_class()` for example

Reply:
Good point, we have to update other modules as well for consistency.